core/src/test/scala/matryoshka/spec.scala

```diff
@@ -175,27 +174,31 @@ object Exp2 {
 }
 
 
-class ExpSpec extends Spec {
+class ExpSpec extends Specification with ScalaCheck with CheckAll {
   import Exp._
 
   implicit val arbExpInt: Arbitrary[Exp[Int]] = arbExp(Arbitrary.arbInt)
   // NB: These are just a sanity check that the data structure created for the
   //     tests is lawful.
-  checkAll(equal.laws[Exp[Int]])
-  checkAll(traverse.laws[Exp])
+  "Exp should satisfy relevant laws" >> {
+    checkAll(equal.laws[Exp[Int]]) append
+    checkAll(traverse.laws[Exp])
+  }
 }
 
-class Exp2Spec extends Spec {
+class Exp2Spec extends Specification with ScalaCheck with CheckAll {
   import Exp2._
 
   implicit val arbExp2Int: Arbitrary[Exp2[Int]] = arbExp2(Arbitrary.arbInt)
   // NB: These are just a sanity check that the data structure created for the
   //     tests is lawful.
-  checkAll(equal.laws[Exp2[Int]])
-  checkAll(functor.laws[Exp2])
+  "Exp2 should satisfy relevant laws" >> {
+    checkAll(equal.laws[Exp2[Int]])
```

Comment:
specs2 made the `Fragments` api more functional by making things `Fragments.foreach` return `Fragments` instead of `Unit` which I think is great. Other things it left as is. I am inclined to squint and call this thing a "dsl" for which mutability was chosen as a technique to implement it. Since I don't like mutability, I would have prefered possibly macros in order to implement the dsl, but I am willing to consider it an implementation detail as it's specs2 responsibility in theory to make sure its mutable dsl works properly.